firestore/integration_test_internal/src/source_test.cc

```diff
@@ -0,0 +1,523 @@
+#include <future>
+
+#include "firebase/firestore.h"
+#include "firestore_integration_test.h"
+
+// These test cases are in sync with native iOS client SDK test
+//   Firestore/Example/Tests/Integration/API/FIRFirestoreSourceTests.mm
+// and native Android client SDK test
+//   firebase-firestore/src/androidTest/java/com/google/firebase/firestore/SourceTest.java
+
+namespace firebase {
+namespace firestore {
+
+using SourceTest = FirestoreIntegrationTest;
+
+TEST_F(SourceTest, TestGetDocumentWhileOnlineWithDefaultGetOptions) {
+  MapFieldValue initialData = {{"key", FieldValue::String("value")}};
+  DocumentReference docRef = DocumentWithData(initialData);
+
+  Future<DocumentSnapshot> future = docRef.Get();
+  WaitFor(future);
+
+  DocumentSnapshot snapshot = *future.result();
+  ASSERT_TRUE(snapshot.exists());
```

Comment:
Is there a reason you're using ASSERT for everything and not EXPECT? ASSERT should really only be used when the test cannot continue if the thing failed.